openraft/src/config/config.rs

```diff
@@ -123,6 +123,16 @@ pub struct Config {
     )]
     pub snapshot_policy: SnapshotPolicy,
 
+    /// Whether to keep all `applied_log`s if they are not included by snapshots.
+    ///
+    /// If your application may rebuild it's state machine from snapshots,
+    /// please set this to true.
+    ///
+    /// By default, `OpenRaft` purges `applied_log`s from time to time regardless of snapshots, because it assumes once
+    /// logs are `applied` to the state machine, logs are persisted on disk. If not so, set this to true.
+    #[clap(long, env = "RAFT_KEEP_UNSNAPSHOTED_LOG", default_value_t = false, value_parser=clap::value_parser!(bool))]
```

Comment:
Is the `value_parser` necessary for it to work?


```suggestion
    #[clap(long, env = "RAFT_KEEP_UNSNAPSHOTED_LOG", default_value_t = false)]
```

Reply:
If not set, the flag would be `true` and tests would fail.

Reply:
I do not know what's going on on your env, but the following snippet gives a `bool` `false` as the default value on my laptop:

```rust
#[derive(Clone, Debug, PartialEq, Eq, Parser)]
#[clap(about, version, author)]
pub struct Config {
    #[clap(long)]
    pub my_flag: bool,
}
impl Default for Config {
    fn default() -> Self {
        <Self as Parser>::parse_from(&Vec::<&'static str>::new())
    }
}

#[tokio::main]
async fn main() {
    let x = Config::default();
    println!("{}", x.my_flag);
}

```

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=70c47c90a49f8e91e86634eca6a86946
Seems like I misunderstood the meaning of `default_value_t`. Should this flag be used along with `value_parser` or it would be set to true?

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=70c47c90a49f8e91e86634eca6a86946
And adding a `value_parser` done the job.

Reply:
> default_value_t

I have not yet used this option. :(